frontend/src/metabase/dashboard/components/RefreshWidget/RefreshWidget.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+import { usePrevious, useUpdateEffect } from "react-use";
+import { t } from "ttag";
+
+import { Box, Popover } from "metabase/ui";
+
+import { RefreshOption } from "./RefreshOption";
+import RefreshWidgetS from "./RefreshWidget.module.css";
+import { RefreshWidgetTarget } from "./RefreshWidgetTarget";
+
+const toSeconds = (minutes: number) => minutes * 60;
+
+const OPTIONS = [
+  { name: t`Off`, period: null },
+  { name: t`1 minute`, period: toSeconds(1) },
+  { name: t`5 minutes`, period: toSeconds(5) },
+  { name: t`10 minutes`, period: toSeconds(10) },
+  { name: t`15 minutes`, period: toSeconds(15) },
+  { name: t`30 minutes`, period: toSeconds(30) },
+  { name: t`60 minutes`, period: toSeconds(60) },
+];
+
+export const RefreshWidget = ({
+  setRefreshElapsedHook,
+  period,
+  onChangePeriod,
+}: {
+  setRefreshElapsedHook?: (hook: (elapsed: number | null) => void) => void;
+  period: number | null;
+  onChangePeriod: (period: number | null) => void;
+}) => {
+  const prevProps = usePrevious({ setRefreshElapsedHook });
+
+  const [elapsed, setElapsed] = useState<number | null>(null);
+
+  const [isOpen, setIsOpen] = useState(false);
+
+  useUpdateEffect(() => {
+    if (
+      setRefreshElapsedHook &&
+      prevProps?.setRefreshElapsedHook !== setRefreshElapsedHook
+    ) {
+      setRefreshElapsedHook((elapsed: number | null) => {
+        setElapsed(elapsed);
+      });
+    }
+  });
```

Comment:
I wonder why we use `usePrevious` in this case instead of using standard `useEffect` to track dependency change on `setRefreshElapsedHook`?

Reply:
I basically copied the code over from JS. Usually when I try to use something like a new `useEffect` in place of the old code something breaks, but I've changed it and things seem to be working correctly still. I've made the change

Reply:
Oh sorry, I thought this was a new logic - forgot for a moment that this is only for TS migration, my bad 🙏🏻 

I'm not exactly sure if `useEffect` would cover all the edge cases though, but if it works fine then ok!